src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -487,15 +521,12 @@ private String getHeaderValueImpl (String uri, String method) {
                 ncstring = zeroPad [len] + ncstring;
         }
 
-        boolean session = algorithm.endsWith ("-SESS");
-        if (session) {
-            algorithm = algorithm.substring(0, algorithm.length() - 5);
-        }
+        boolean session = algorithm.endsWith ("-sess");
```

Comment:
should that be `digest.endsWith("-sess");` ?

Reply:
No. The new method `setAlgorithmNames` ensures that the suffix is in lower case in the parameters object.

Reply:
No, the digest field refers to the actual message digest algorithm (as known to the security libraries). The algorithm field holds the algorithm name as it is defined in  RFC71616.

Reply:
I am confused here - because you converted `algorithm` to upper case, so it should never end with `-sess`?

Reply:
Look at line 478: The `algorithm` field is reset here to be the upper case of the digest name plus the -sess suffix in lower case.

Reply:
Doh! Somehow I'd read that as `digestName = digestName + "-sess"`.
All clear now.